aptos-move/aptos-vm-types/src/change_set.rs

```diff
@@ -60,71 +175,167 @@ impl VMChangeSet {
         Ok(change_set)
     }
 
-    pub fn write_set(&self) -> &WriteSet {
-        &self.write_set
+    /// Returns a new change set, built from the storage representation. Note
+    /// that this is a rather expensive operation because it re-splits resources
```

Comment:
Great Warning about the complexity.  Maybe even rewrite such that the "WARNING" is clearly visible.